vj4/model/adaptor/problem.py

```diff
@@ -45,6 +45,18 @@ def get_categories():
   await domain.inc_user(domain_id, owner_uid, num_problems=1)
   return pid
 
+async def copy(pdoc, dest_domain_id: str, owner_uid: int,
+               pid: document.convert_doc_id=None, hidden: bool=False):
+  data = pdoc['data']
+  if type(data) is objectid.ObjectId:
```

Comment:
better to add a comment indicating that this is the case that data is copied / linked from another problem.